Source/WebCore/rendering/RenderImage.cpp

```diff
@@ -410,7 +410,9 @@ void RenderImage::repaintOrMarkForLayout(ImageSizeChangeType imageSizeChange, co
         if (rect) {
             // The image changed rect is in source image coordinates (pre-zooming),
             // so map from the bounds of the image to the contentsBox.
-            repaintRect.intersect(enclosingIntRect(mapRect(*rect, FloatRect(FloatPoint(), imageResource().imageSize(1.0f)), repaintRect)));
+            auto srcImage = imageResource().image(flooredIntSize(contentBoxSize()));
```

Comment:
How expensive is this?

Reply:
negligible, we should be reading cached layout values here. fwiw, a similar call is made in `paintReplaced`